music21/environment.py

```diff
@@ -517,15 +525,25 @@ def getDefaultRootTempDir(self):
 
         # this returns the root temp dir; this does not create a new dir
         self.defaultRootTempDir = pathlib.Path(tempfile.gettempdir()) / 'music21'
-        # if this path already exists, we have nothing more to do
-        if self.defaultRootTempDir.exists():
+
+        # if this path already exists, readable and writable, we have nothing more to do
+        readable = os.access(self.defaultRootTempDir, os.R_OK)
+        writable = os.access(self.defaultRootTempDir, os.W_OK)
+        if self.defaultRootTempDir.exists() and readable and writable:
             return self.defaultRootTempDir
         else:
             # make this directory as a temp directory
             try:
                 self.defaultRootTempDir.mkdir()
             except OSError:
-                self.defaultRootTempDir = pathlib.Path(tempfile.mkdtemp(prefix="music21-"))
+                if sys.platform.startswith('linux'):
+                    uid = os.getuid()
+                    dir_path = pathlib.Path(tempfile.gettempdir()) / f'music21-{uid}'
```

Comment:
Ah, is this where we were creating too many directories? Each time User X lacks write access to the root temp dir, we catch `OSError` and then attempt to create more directories with new hashes each time? Makes sense to change this. But is there a reason you didn't want to use `prefix=` like below? I just feel like we should have both branches behave the same way, either construct the tempdir inline, or construct a path and then mkdir() after.

Reply:
> Ah, is this where we were creating too many directories? Each time User X lacks write access to the root temp dir, we catch OSError and then attempt to create more directories with new hashes each time?

Yes. It creates a new directory for each different python session.

> But is there a reason you didn't want to use prefix= like below? I just feel like we should have both branches behave the same way, either construct the tempdir inline, or construct a path and then mkdir() after.

I have tried to make code here more elegant. But tempfile doesn't provide suitable API to make both branches behave in the same way. Maybe I can change it to
```
...
dir_path = pathlib.Path(tempfile.gettempdir()) / f'music21-{uid}'
dir_path.mkdir()
...
```
and remove the if block
```
...
if not dir_path.exists():
    dir_path.mkdir()
...
```

Reply:
Sure. I was just wondering why we weren't using `tempfile.mkdtemp()`. I know `NamedTemporaryDirectory` is preferred over mkdtemp, but I wasn't sure why `gettempdir()` and then `mkdir()` is superior to just mkdtemp. Honestly just wondering if you had a reason or if I'm missing something.

Reply:
The reason is `tempfile.mkdtemp()` create a directory with random name (even called with `prefix`). It can't be chased in different python session. If program can't get the last-created temp directory, then it have to create a new one. There are two solution as I know. The first one is storing the name of temp directory somewhere in the disk, and the second one is using a fixed name. I prefer the second one so I use `gettempdir()+suffix_name` and then `mkdir()`.

Reply:
Thanks, that helps me understand! All good.